### ISSUE TITLE
Netflix reports code M7121-1331 on ungoogled-chromium >=113

### ISSUE THREAD

**Issue by LoganDark** [ungoogled-software/ungoogled-chromium-windows#247]
On my computer, Netflix used to work with I believe ungoogled-chromium 112, however on 113 and 114, installing the latest Widevine (4.10.2557.0 in my case) doesn't seem to work, and Netflix reports the error M7121-1331, which is a generic "something went wrong".

**Comment by PF4Public** [comment]
You need to provide more information so that we could help you.

**Comment by LoganDark** [comment]
> You need to provide more information so that we could help you.

Like what? what information do you need?

Here's my current Chrome and OS version as reported in about:version:

```
114.0.5735.134 (Official Build, ungoogled-chromium) (64-bit)
Revision | 960a08e3d10d3a6911d6e7f4a38b2c73f4aa0fa3-refs/branch-heads/5735@{#1271}
OS | Windows 11 Version 22H2 (Build 22621.1485)
```

(updated since opening the issue, but still have the same problem)

I followed the guide [here](https://chromium.woolyss.com/#widevine) for installing Widevine, and it shows as installed on chrome://components:

![image](https://github.com/ungoogled-software/ungoogled-chromium-windows/assets/4723091/3eb8c370-9e0c-433b-95f3-b0b890d3c4e0)

The Widevine demos at https://shaka-player-demo.appspot.com all work properly. Disney+ can also play DRM-protected content perfectly fine. The flags I have enabled are:

`--custom-ntp=chrome://new-tab-page --disable-search-engine-collection --disable-sharing-hub --enable-gpu-rasterization --enable-zero-copy --extension-mime-request-handling=always-prompt-for-install --hide-sidepanel-button --ignore-gpu-blocklist --keep-old-history --remove-grab-handle --remove-tabsearch-button --disable-smooth-scrolling --use-angle=gl --enable-features=CanvasOopRasterization,DisableQRGenerator,ElasticOverscroll,EnableDrDc,OverlayScrollbar,SystemNotifications --disable-features=HardwareMediaKeyHandling,OverscrollHistoryNavigation,Vulkan`

Please let me know what information I should provide to help you.

**Comment by PF4Public** [comment]
Inspecting the console output of developer tools at the failing page might be helpful

**Comment by LoganDark** [comment]
I tried that, along with correlating network requests with the error page.

The console had these:

```
otSDKStub.js:1     Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
POST https://www.netflix.com/msl/playapi/cadmium/logblob/1?reqAttempt=2&reqName=logblob&clienttype=akira&uiversion=vc5ef9af4&browsername=chrome&browserversion=114.0.0.0&osname=windows&osversion=10.0 net::ERR_BLOCKED_BY_CLIENT
/msl/playapi/cadmium/logblob/1?reqAttempt=1&reqName=logblob&clienttype=akira&uiversion=vc5ef9af4&browsername=chrome&browserversion=114.0.0.0&osname=windows&osversion=10.0:1     Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
/msl/playapi/cadmium/logblob/1?reqAttempt=2&reqName=logblob&clienttype=akira&uiversion=vc5ef9af4&browsername=chrome&browserversion=114.0.0.0&osname=windows&osversion=10.0:1     Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
/msl/playapi/cadmium/logblob/1?reqAttempt=1&reqName=logblob&clienttype=akira&uiversion=vc5ef9af4&browsername=chrome&browserversion=114.0.0.0&osname=windows&osversion=10.0:1     Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
/msl/playapi/cadmium/logblob/1?reqAttempt=2&reqName=logblob&clienttype=akira&uiversion=vc5ef9af4&browsername=chrome&browserversion=114.0.0.0&osname=windows&osversion=10.0:1     Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
```

The first is cookielaw.org bloat, and the rest I think are just analytics endpoints.

In the network requests we have:

- a few  `https://oca-api.us-east-2.origin.prodaa.netflix.com/ftl/probe?data=5120&device=web&ord=3&recipe=unifiedProbe&probe=us-east-2&pulse=2` completing successfully, to locate the video stream
- some various requests to URLs like `https://ipv6-c056-pdx001-ix.1.oca.nflxvideo.net/range/240220250-240636335?o=1&v=61&e=1687601772&t=[censored]&sc=[censored]` to retrieve the video data
- `https://www.netflix.com/nq/msl_v1/cadmium/pbo_licenses/%5E1.0.0/router?reqAttempt=2&reqName=license&clienttype=akira&uiversion=vc5ef9af4&browsername=chrome&browserversion=114.0.0.0&osname=windows&osversion=10.0` to presumably locate some license? regardless, also success
- a failed attempt to that `https://www.netflix.com/msl/playapi/cadmium/logblob/1?...` URL
- error screen shows up

can you test Netflix and see if it works for you? if it does, then I could probably check extensions and etc., but if it's also broken for you then it's probably an ungoogled-chromium issue

**Comment by PF4Public** [comment]
I don't have Netflix account to test.

You could test empty profile without extensions and with default flags with the help of `--user-data-dir`. You also did not provide any information about how you installed ungogled-chromium.

**Comment by LoganDark** [comment]
> You also did not provide any information about how you installed ungogled-chromium.

Sorry. I used the EXE installer from [the latest release](https://github.com/ungoogled-software/ungoogled-chromium-windows/releases/tag/114.0.5735.134-1.1). Thought it would be obvious under the repository for a specific distribution that I am using that distribution.

> You could test empty profile without extensions and with default flags with the help of `--user-data-dir`.

Ok, I'll try.

**Comment by PF4Public** [comment]
> Sorry. I used the EXE installer (x64) from [the latest release](https://github.com/ungoogled-software/ungoogled-chromium-windows/releases/tag/114.0.5735.134-1.1). Thought it would be obvious under the repository for a specific distribution that I am using that distribution.

Unfortunately not. We are constantly bombarded by users using third-party binaries which they get outside this repository, outside this organization, provided by unrelated people with problems we possibly cannot solve.

**Comment by LoganDark** [comment]
![image](https://github.com/ungoogled-software/ungoogled-chromium-windows/assets/4723091/ab4da296-da69-4c9d-ba50-2bd9e3158997)

It does occur on an empty profile with no flags.

**Comment by PF4Public** [comment]
Did you use Chromium's builtin profile feature or separate temporary folder via `--user-data-dir`?

**Comment by LoganDark** [comment]
> Did you use Chromium's builtin profile feature or separate temporary folder via `--user-data-dir`?

Separate temporary folder. As you can see no other profiles exist. The command line I used was `"C:\Users\LoganDark\AppData\Local\Chromium\Application\chrome.exe" --user-data-dir=C:\Chrometest`, after ensuring that folder did not exist.

**Comment by PF4Public** [comment]
Are there any changes in console? Maybe you have blocked something via hosts?

**Comment by LoganDark** [comment]
> Maybe you have blocked something via hosts?

I block things with a [web application firewall](https://safing.io) that logs when things are blocked (so more useful than a hosts file!), but logs confirm nothing from Chromium was blocked.

![image](https://github.com/ungoogled-software/ungoogled-chromium-windows/assets/4723091/e0572269-f053-427e-9ff0-ba020cbdcaf2)

> Are there any changes in console?

I don't see any BLOCKED_BY_CLIENT errors because that profile has no extensions. But Netflix still shows the error page, so that doesn't seem to have had an impact.

**Comment by PF4Public** [comment]
Then I'm out of ideas. Could be Netflix changed something internal and tries blocking suspicious (according to them) browsers or I don't know.

Could be that specific title at issue? Maybe others work fine?

Try vanilla Chromium. Maybe it is more general issue?

**Comment by LoganDark** [comment]
> Could be that specific title at issue? Maybe others work fine?

No, tried quite a few.

> Try vanilla Chromium. Maybe it is more general issue?

Possibly. I tried changing user-agent to chrome's and it didn't work. Regardless, I don't know if I'm comfortable running googled chromium right now.

**Comment by pomah86** [comment]
I have the same issue with netflix but aslo with whatsapp, when running https://web.whatsapp.com/ it now wont load the qr code to allow you to connect a device (works just fine on the same computer using e.g. edge)

Not sure if these are connected but seems odd that they both started just now...

**Comment by PF4Public** [comment]
> but aslo with whatsapp, when running https://web.whatsapp.com/ it now wont load the qr code to allow you to connect a device (works just fine on the same computer using e.g. edge)

Have you tried empty profile? This could happen if you disable Javascript JIT, but AFAIK we do not ship this feature in ungoogled-chromium windows, which means you used third party binary for which we cannot provide support.

**Comment by teeminus** [comment]
WhatsApp Web works fine for me with 114.0.5735.199-1.1 in incognito mode.

**Comment by pomah86** [comment]
Tried incognito but same issue for me, wounder what that could be then?

**Comment by teeminus** [comment]
Have you tried deleting the application data manually?

**Comment by pomah86** [comment]
> Have you tried deleting the application data manually?

Not really sure what you mean it is run in the web tab, what data is there to delete except for cookies and such (that I have deleted, and anything associated with that site)

**Comment by teeminus** [comment]
I was talking about the application data stored in '%appdata%' (not sure about the exact location). Deleting that folder is like a clean install of chromium.

**Comment by teeminus** [comment]
Please note that this will delete all stored credentials, plugins etc.

**Comment by pomah86** [comment]
Ok, I will try that later, most things are saved in bitwarden anyways, just need to export the bookmarks

**Comment by 0xGingi** [comment]
The folder should be  %localappdata%/Chromium/User Data, the Application folder should be the browser itself, so you could try deleting the Chromium folder entirely and reinstalling the browser if removing just the User Data folder does not work

**Comment by pomah86** [comment]
Hi, deleted the entire chromium folder as suggested above, installed a fresh install, no plugins, no settings done, just went to https://web.whatsapp.com/ and the page just continues to reload the QR code, but never stops.

I then allowed, webgl, javascript, 3rd party cockies, everything that was blocked under "web settings "for the specific site I unblocked with the same result as before. Opening the same link in e.g. edge on the same computer works just fine.

This also worked for me until version 113 or so.

To be completely transparent, I'm using builds from: https://chromium.woolyss.com/, on win10, if that makes any difference.

**Comment by pomah86** [comment]
I get the following "issues" with dev console:

Audit usage of navigator.userAgent, navigator.appVersion, and navigator.platform

`A page or script is accessing at least one of navigator.userAgent, navigator.appVersion, and navigator.platform. Starting in Chrome 101, the amount of information available in the User Agent string will be reduced.
To fix this issue, replace the usage of navigator.userAgent, navigator.appVersion, and navigator.platform with feature detection, progressive enhancement, or migrate to navigator.userAgentData.
Note that for performance reasons, only the first access to one of the properties is shown.`


Console also gives me this:
![Skärmbild 2023-07-03 091612](https://github.com/ungoogled-software/ungoogled-chromium-windows/assets/7014789/f0059838-bc87-4302-8eff-d8fafb704c2c)

**Comment by teeminus** [comment]
> To be completely transparent, I'm using builds from: https://chromium.woolyss.com/, on win10, if that makes any difference.

Yes, that actually makes a huge difference. We do not publish our releases there, the chromium you are using is not build from this project. We only provide support for our releases.

**Comment by pomah86** [comment]
Well, did I clean install again, using the relase from this github and it all works, as it should. Netflix is still giving me the same issue, the error code M7701-1003, 

1 error:
![image](https://github.com/ungoogled-software/ungoogled-chromium-windows/assets/7014789/9514a26b-705d-4f1d-8666-4cc5a8df8230)

and 1 message:
![image](https://github.com/ungoogled-software/ungoogled-chromium-windows/assets/7014789/65d210e2-b702-4d6e-80b4-3d679ac2d47f)

**Comment by PF4Public** [comment]
Come on guys! Why no one noticed [my comment above](https://github.com/ungoogled-software/ungoogled-chromium-windows/issues/247#issuecomment-1616638726)?

**Comment by xTr07a** [comment]
Any updates regarding this issue? I am using ungoogled-chromium version 115.0.5790.99-1.1 on windows and have the same problem.

**Comment by PF4Public** [comment]
> Any updates regarding this issue?

DRM features are alien for ungoogled-chromium goals. Besides that, in order to properly debug this issue, one needs to have an account with those closed platforms. These two factors result in us having tons of similar issues without resolution. So until someone volunteers to investigate those issues, no progress is to be expected. And on top of that, chances are that proper fix for this would contradict the goals of ungoogled-chromium and won't be accepted into this repository.

**Comment by pomah86** [comment]
I can just add that this is on latest ungoogled version (115.0.5790.110): 
![image](https://github.com/ungoogled-software/ungoogled-chromium-windows/assets/7014789/9c989c25-6745-4a9e-aaba-a42b989da9cf)

When looking at my drm setttings I have this: 
![image](https://github.com/ungoogled-software/ungoogled-chromium-windows/assets/7014789/d03d6dbb-f2e3-44ce-95fb-642d60061f19)

So all seems OK, but still not working :(

**Comment by xTr07a** [comment]
> I can see that that folder is empty.
> 
> So my conclusion is that I do not have cdm installed?

Try this: https://ungoogled-software.github.io/ungoogled-chromium-wiki/faq#how-do-i-install-widevine-cdm

**Comment by TorutheRedFox** [comment]
> > I can see that that folder is empty.
> > So my conclusion is that I do not have cdm installed?
> 
> Try this: https://ungoogled-software.github.io/ungoogled-chromium-wiki/faq#how-do-i-install-widevine-cdm

a lot of streaming services don't work regardless of whether widevine is installed or not

**Comment by pomah86** [comment]
> > I can see that that folder is empty.
> > So my conclusion is that I do not have cdm installed?
> 
> Try this: https://ungoogled-software.github.io/ungoogled-chromium-wiki/faq#how-do-i-install-widevine-cdm

Funny outcome :)

Netflix starts to play but after 1 sec give out the same error as before "m7121-1331", "Please try another title"

But amazon prime does not and plays without an issue, but after reading up I guess widewine is not part of this project due to it being closed source, and widewine is also the reason netflix wont play, so I gues for the sake of this thread we can close it down and the reason for the issue is lack of widewine support natively?

**Comment by ColumbusAI** [comment]
i installed the most recent build of ungoogled chromium two days ago, followed the faq to paste the widevine files in according to the directory structure, and still have the same issue with Netflix 'cannot play instant title' from screenshots above.

No drm media will play though Widevine shows as loaded. Anyone figure this out??

**Comment by pomah86** [comment]
I can only make it play on amazon prime, netflix, hbo does not work.

**Comment by ColumbusAI** [comment]
Just read a post by dev for Thorium browser and he explains why it won't work. Oh well ill just have to use edge for netflix

**Comment by ThreeDeeJay** [comment]
Wow it's been a couple years already. Did anyone manage to find a solution for this or a different open source browser?
I followed the guide to install Widevine which works in the test sites.
But I'm having no luck with Netflix playback on Ungoogled Chromium or Cromite and wonder if Brave works because I really don't wanna have to install Chrome or Edge.
I'm getting both the playback error and it even signs me out with a fresh install. 

EDIT: Apparently the problem is explained here:
- https://github.com/ungoogled-software/ungoogled-chromium/issues/1566#issuecomment-880734483
- https://github.com/Alex313031/thorium/blob/main/docs/FAQ.md
